ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -355,20 +360,30 @@ protected void start() {
     b.group(group).channel(NioSocketChannel.class).handler(new ChannelInitializer<SocketChannel>() {
       @Override
       public void initChannel(SocketChannel ch) throws Exception {
+        channelConnectLatch.await();
         if (sslFactory != null) {
+          String host = channelToHost.get(ch);
+          if (host == null) {
+            throw new Exception("Unrecognized host!");
+          }
+          logger.info("Initializing the channel to {}", host);
```

Comment:
this if statement and logging could be moved to the chnnalConnectListener so we can get rid of the latch wait here.

Reply:
resolved by offline discussion